addons/mail/static/src/js/activity.js

```diff
@@ -714,7 +714,7 @@ var KanbanActivity = BasicActivity.extend({
     _renderDropdown: function () {
         var self = this;
         this.$('.o_activity')
-            .toggleClass('dropdown-menu-right', config.device.isMobile)
+            .toggleClass('dropdown-menu-right', config.device.isMobile || self.$('.dropdown-toggle').offset()['left'] + self.$('.o_activity').outerWidth() > window.innerWidth)
```

Comment:
This line becomes a bit convoluted. How about putting the new part of the condition in a dedicated function, with a good, self-explanatory name? This way, it would save us from having to run the code in our heads to understand what it does, and it would better convey the intention you had in writing the code.